ml-agents/mlagents/trainers/policy/torch_policy.py

```diff
@@ -69,18 +70,26 @@ def __init__(
             "Losses/Value Loss": "value_loss",
             "Losses/Policy Loss": "policy_loss",
         }
-        self.actor_critic = ActorCritic(
-            observation_shapes=self.behavior_spec.observation_shapes,
-            network_settings=trainer_settings.network_settings,
-            act_type=behavior_spec.action_type,
-            act_size=self.act_size,
-            stream_names=reward_signal_names,
-            separate_critic=separate_critic
-            if separate_critic is not None
-            else self.use_continuous_act,
-            conditional_sigma=self.condition_sigma_on_obs,
-            tanh_squash=tanh_squash,
-        )
+        if separate_critic:
```

Comment:
I think we can grab the right class constructor in the if/then statement, and then use it to instantiate the class after. That way we save a bunch of lines.

Reply:
````
if separate_critic:
    ac_class = SeparateActorCritic
else:
    ac_class = ActorCritic

Reply:
Done